glass/fields.py

```diff
@@ -218,10 +218,18 @@ def multalm(
         The product of alm and bl.
 
     """
-    n = len(bl)
     out = np.asanyarray(alm) if inplace else np.copy(alm)
-    for m in range(n):
-        out[m * n - m * (m - 1) // 2 : (m + 1) * n - m * (m + 1) // 2] *= bl[m:]
+    n = len(bl)
+    for d in range(n):
+        indices = []
+        for m in range(d, n):
+            # dynamically calculate indices
+            index = m * (m + 1) // 2 + d
+            indices.append(index)
+
+        if indices:
+            out[indices] *= bl[: len(indices)]
```

Comment:
The function now assumes that the incoming `alm`s are in the order -
```
[00, 10, 11, 20, 21, 22, 30, 31, 32, 33, ...]
```

Therefore, it now multiplies `alm` with `bl` in the following way -
```
bl := [1, 10, 100, 1000]

[0, 10, 20, 30] * bl → [0, 100, 2000, 30000]
[11, 21, 31] * bl[:3] → [11, 210, 3100]
[22, 32] * bl[:2] → [22, 320]
[33] * bl[:1] → [33]
```

Reply:
Ah, I got the multiplication wrong! That's why I was confused about why we are switching to this order. Thanks for the correction!